magento-research/pwa-studio
# [dev]: Add eslint plugin for react hooks

## Description
We use hooks now. We don't lint the hooks. We should lint the hooks. I added released a new version of the `magento-eslint` repo, updated the version here, added the requisite plugin, and ran it once to fix errors (the only one was in `app.js`). In this PR I also fixed the warnings. There were two cases around `addToast` where we did not update the dependency according to the warning. These are potential bugs and should be addressed with this issue: #1377 

## Related Issue





Closes #1357.

## Verification Steps
Run `yarn run lint` and make sure there are no errors.
Also test:
* Offline/Online (useToasts)
* Error handling (useToasts), throw an error in miniCart.js#render
* searchBar (useDropdown)
* Thumbnail (useWindowSize)
* autocomplete (useQuery)
* category pages (useQuery)

Things tested inherently:
* useEventListener, tested by useWindowSize and useDropdown

## Screenshots / Screen Captures (if appropriate):

![](https://i.gyazo.com/7f594d4c82d79fcb922e3f1435c8dec3.png)
## Proposed Labels for Change Type/Package




## Checklist:
